File: adventure-engine/frontend/src/services/geocode.ts
import { apiGet } from './api';

export interface GeocodeResult {
  latitude: number;
  longitude: number;
  label: string;
  country: string | null;
}

export function resolveLocation(query: string): Promise<GeocodeResult> {
  return apiGet<GeocodeResult>('/geocode/', { query });
}

File: adventure-engine/backend/app/api/routes/geocode.py
from fastapi import APIRouter, HTTPException, Query

from app.schemas.geocode import GeocodeResultRead
from app.services.geocoding import GeocodeError, geocode

router = APIRouter(prefix="/geocode", tags=["geocode"])


@router.get("/", response_model=GeocodeResultRead)
def resolve_location(
    query: str = Query(..., min_length=1, description="A city or airport name, e.g. 'Chicago' or 'JFK airport'"),
) -> GeocodeResultRead:
    try:
        result = geocode(query)
    except GeocodeError as exc:
        raise HTTPException(status_code=404, detail=str(exc))

    return GeocodeResultRead(
        latitude=result.latitude,
        longitude=result.longitude,
        label=result.label,
        country=result.country,
    )
